docs/assembly.rst

```diff
@@ -394,6 +394,16 @@ use ``x_slot``, and to retrieve the byte-offset you use ``x_offset``.
 
 Local Solidity variables are available for assignments, for example:
 
+.. warning::
+    If you access variables of a type that spans less than 256 bits
+    (for example ``uint64``, ``address``, ``bytes16`` or ``byte``),
+    you cannot make any assumptions about bits not part of the
+    encoding of the type. Especially, do not assume them to be zero.
+    To be safe, always clear the data properly before you use it
+    in a context where this is important:
+    ``uint32 x = f(); assembly { x := and(x, 0xffffffff) /* now use x */ }``
+    To clean signed types, you can use the ``signextend`` opcode.
```

Comment:
Would it be possible to provide an example for the use of `signextend` to clear higher order bits?

Reply:
I've added an example. Is this correct and would it be what you've requested?